src/modules/Auth/services/auth.ts

```diff
@@ -1,26 +1,63 @@
-export interface User {
-  name: string;
-  email: string;
-  password?: string;
+import Keycloak from 'keycloak-js';
+
+const isProduction = process.env.NODE_ENV === 'production';
+const configUrl = isProduction ? 'keycloak.json' : 'keycloak.dev.json';
+const onLoad: Keycloak.KeycloakOnLoad = 'login-required';
+const _kc = Keycloak(configUrl);
+
+function init(appEntrypoint: () => void): void {
+  _kc
+    .init({
+      onLoad,
+    })
+    .then((auth) => {
+      if (!auth) {
+        window.location.reload();
+      } else {
+        _kc
+          .loadUserProfile()
+          .then(() => {
```

Comment:
could be

```
_kc.loadUserProfile()
            .then(appEntrypoint())
```

Reply:
Actually, I changed to use async/await syntax